pkg/registry/authorization/selfsubjectaccessreview/rest.go

```diff
@@ -72,6 +76,33 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 		}
 	}
 
+	hasUserFull, hasUserCheckAccess := false, false
+	for _, scope := range userFromCtx.GetExtra()[authorizationv1.ScopesKey] {
+		switch scope {
+		case authorizationscope.UserFull:
+			hasUserFull = true
+			break
+
+		case authorizationscope.UserAccessCheck:
+			hasUserCheckAccess = true
+		}
+	}
+
+	var userToCheck user.Info
+	if hasUserCheckAccess && !hasUserFull {
+		userToCheck = &user.DefaultInfo{
+			Name:   userFromCtx.GetName(),
+			UID:    userFromCtx.GetUID(),
+			Groups: userFromCtx.GetGroups(),
+			Extra: map[string][]string{
+				authorizationv1.ScopesKey: {authorizationscope.UserFull},
```

Comment:
what about any other possible extras that we do not own?

Reply:
Good point -- fixed, it now copies all extras before modifying scopes.